dask/tests/test_core.py

```diff
@@ -45,62 +41,140 @@ def test_preorder_traversal():
 
-    result = getm(d, ':z', get=getm)
-    assert result == 3
+    def test_get_with_list(self):
+        d = {'x': 1, 'y': 2, 'z': (sum, ['x', 'y'])}
 
-    assert contains(cache, {(':x',): 1,
-                            (':y',): 2,
-                            (':z',): 3})
+        assert self.get(d, ['x', 'y']) == (1, 2)
+        assert self.get(d, 'z') == 3
 
-def test_data_not_in_dict_is_ok():
-    d = {'x': 1, 'y': (add, 'x', 10)}
-    assert get(d, 'y') == 11
+    def test_get_with_nested_list(self):
+        d = {'x': 1, 'y': 2, 'z': (sum, ['x', 'y'])}
 
+        assert self.get(d, [['x'], 'y']) == ((1,), 2)
+        assert self.get(d, 'z') == 3
 
-def test_get_with_list():
-    d = {'x': 1, 'y': 2, 'z': (sum, ['x', 'y'])}
+    def test_get_works_with_unhashables_in_values(self):
+        f = lambda x, y: x + len(y)
+        d = {'x': 1, 'y': (f, 'x', set([1]))}
 
-    assert get(d, ['x', 'y']) == [1, 2]
-    assert get(d, 'z') == 3
+        assert self.get(d, 'y') == 2
 
+    def test_get_laziness(self):
+        def isconcrete(arg):
+            return isinstance(arg, list)
 
-def test_get_with_nested_list():
-    d = {'x': 1, 'y': 2, 'z': (sum, ['x', 'y'])}
+        d = {'x': 1, 'y': 2, 'z': (isconcrete, ['x', 'y'])}
 
-    assert get(d, [['x'], 'y']) == [[1], 2]
-    assert get(d, 'z') == 3
+        assert self.get(d, ['x', 'y']) == (1, 2)
+        assert self.get(d, 'z') == False
 
+    def test_nested_tasks(self):
+        d = {'x': 1,
+             'y': (inc, 'x'),
+             'z': (add, (inc, 'x'), 'y')}
 
-def test_get_works_with_unhashables_in_values():
-    f = lambda x, y: x + len(y)
-    d = {'x': 1, 'y': (f, 'x', set([1]))}
+        assert self.get(d, 'z') == 4
 
-    assert get(d, 'y') == 2
+    def test_get_stack_limit(self):
+        d = dict(('x%s' % (i+1), (inc, 'x%s' % i)) for i in range(10000))
+        d['x0'] = 0
+        assert self.get(d, 'x10000') == 10000
+        # introduce cycle
+        d['x5000'] = (inc, 'x5001')
 
+        try:
+            self.get(d, 'x10000')
+        except (RuntimeError, ValueError) as e:
+            if isinstance(e, RuntimeError):
+                assert str(e) == 'Cycle detected in Dask: x5001->x5000->x5001'
+            elif isinstance(e, ValueError):
+                assert str(e).startswith('Found no accessible jobs in dask')
+        else:
+            msg = 'dask with infinite cycle should have raised an exception.'
+            self.assertTrue(False, msg=msg)
 
-def test_get_laziness():
-    def isconcrete(arg):
-        return isinstance(arg, list)
+        assert self.get(d, 'x4999') == 4999
 
-    d = {'x': 1, 'y': 2, 'z': (isconcrete, ['x', 'y'])}
 
-    assert get(d, ['x', 'y']) == [1, 2]
-    assert get(d, 'z') == False
+def test_quote(self):
+        literals = [[1, 2, 3], (add, 1, 2),
+                    [1, [2, 3]], (add, 1, (add, 2, 3))]
+
+        for l in literals:
+            assert self.get({'x': quote(l)}, 'x') == l
+
+
+def test_TestGet_class():
```

Comment:
This one is a test function for the TestGet class.  Perhaps a bit too much, but at least shows how it should be used when imported.